simd/src/Kokkos_Simd_Common.hpp

```diff
@@ -0,0 +1,262 @@
+    simd<T, Abi>& a, Kokkos::Impl::identity_t<simd<T, Abi>> const& b) {
+  a = a + b;
+  return a;
+}
+
+template <class T, class Abi>
+KOKKOS_FORCEINLINE_FUNCTION simd<T, Abi>& operator-=(
+    simd<T, Abi>& a, Kokkos::Impl::identity_t<simd<T, Abi>> const& b) {
+  a = a - b;
+  return a;
+}
+
+template <class T, class Abi>
+KOKKOS_FORCEINLINE_FUNCTION simd<T, Abi>& operator*=(
+    simd<T, Abi>& a, Kokkos::Impl::identity_t<simd<T, Abi>> const& b) {
+  a = a * b;
+  return a;
+}
+
+template <class T, class Abi>
+KOKKOS_FORCEINLINE_FUNCTION simd<T, Abi>& operator/=(
+    simd<T, Abi>& a, Kokkos::Impl::identity_t<simd<T, Abi>> const& b) {
+  a = a / b;
+  return a;
+}
+
+template <class U, class T, class Abi>
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION
+    std::enable_if_t<std::is_arithmetic_v<U>, simd<T, Abi>>
+    operator+(U const& a, simd<T, Abi> const& b) {
+  return simd<T, Abi>(T(a)) + b;
+}
+
+template <class U, class T, class Abi>
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION
+    std::enable_if_t<std::is_arithmetic_v<U>, simd<T, Abi>>
+    operator+(simd<T, Abi> const& a, U const& b) {
+  return a + simd<T, Abi>(T(b));
+}
+
+template <class U, class T, class Abi>
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION
+    std::enable_if_t<std::is_arithmetic_v<U>, simd<T, Abi>>
+    operator-(U const& a, simd<T, Abi> const& b) {
+  return simd<T, Abi>(T(a)) - b;
+}
+
+template <class U, class T, class Abi>
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION
+    std::enable_if_t<std::is_arithmetic_v<U>, simd<T, Abi>>
+    operator-(simd<T, Abi> const& a, U const& b) {
+  return a - simd<T, Abi>(T(b));
+}
+
+template <class U, class T, class Abi>
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION
+    std::enable_if_t<std::is_arithmetic_v<U>, simd<T, Abi>>
+    operator*(U const& a, simd<T, Abi> const& b) {
+  return simd<T, Abi>(T(a)) * b;
+}
+
+template <class U, class T, class Abi>
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION
+    std::enable_if_t<std::is_arithmetic_v<U>, simd<T, Abi>>
+    operator*(simd<T, Abi> const& a, U const& b) {
+  return a * simd<T, Abi>(T(b));
+}
+
+template <class U, class T, class Abi>
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION
+    std::enable_if_t<std::is_arithmetic_v<U>, simd<T, Abi>>
+    operator/(U const& a, simd<T, Abi> const& b) {
+  return simd<T, Abi>(T(a)) / b;
+}
+
+template <class U, class T, class Abi>
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION
+    std::enable_if_t<std::is_arithmetic_v<U>, simd<T, Abi>>
+    operator/(simd<T, Abi> const& a, U const& b) {
+  return a / simd<T, Abi>(T(b));
+}
+
+template <class U, class T, class Abi>
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION
+    std::enable_if_t<std::is_arithmetic_v<U>, simd<T, Abi>&>
+    operator/=(simd<T, Abi>& a, U const& b) {
+  a = a / b;
+  return a;
+}
+
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION constexpr bool all_of(bool a) {
+  return a;
+}
+
+[[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION constexpr bool any_of(bool a) {
+  return a;
+}
```

Comment:
What do you need these for?

Reply:
This is actually really important (and specified in the ISO TS) so that I can write code like this:
```cpp
template <class T>
void do_expensive_physics_if(T should_do)
{
  if (any_of(should_do)) {
    // if any of the lanes are true then do lots of vectorized math
    // but if none of the lanes are true then we can skip this work
  }
}
```
And it works when `T=bool` and when `T=simd_mask<...>`, basically generic code that works with SIMD types or regular arithmetic types